pandas/tests/io/excel/test_readers.py

```diff
@@ -1278,6 +1278,13 @@ def test_ignore_chartsheets_by_int(self, request, read_ext):
         ):
             pd.read_excel("chartsheet" + read_ext, sheet_name=1)
 
+    def test_dtype_dict(self, read_ext):
+        filename = "test_common_headers" + read_ext
+        dtype_dict = {"a": str, "b": str, "c": str}
+        dtype_dict_copy = dtype_dict.copy()
+        pd.read_excel(filename, dtype=dtype_dict)
+        assert dtype_dict == dtype_dict_copy, "dtype dict changed"
```

Comment:
Can we also check that the resulting frame is as expected? (I know this is focusing on the dtypes dict, but may as well also test the reading portion here since unlikely we have great coverage for `dtypes` dict with duplicate cols)